sa/storage-authority_test.go

```diff
@@ -133,6 +134,10 @@ func TestAddAuthorization(t *testing.T) {
 
 	dbPa, err = sa.GetAuthorization(PA.ID)
 	test.AssertNotError(t, err, "Couldn't get authorization with ID "+PA.ID)
+
+	authz, err := sa.GetLatestValidAuthorization(0, identifier)
```

Comment:
Please add a separate test case for this function. I know our codebase doesn't set the best example. We have lots of very long test functions. But we'd like to make that better. :-)

Also, I'd like to see more things tested:
 - what happens when there are multiple authorizations
 - what happens when one is invalid
 - what happens when there is a newer authorization that is invalid and an older, valid one
 - what happens when there are no authorizations

Ideally each of these would be a separate function, which may mean abstracting away some of the necessary test setup into its own function.